### ISSUE TITLE
Use GNI remote events for remote counters

### ISSUE THREAD

**Issue by ztiffany** [ofi-cray/libfabric-cray#869]
Remote events could be used to signal a peer to perform a FI_RMA_EVENT counter increment.  Remote events may be lost if the remote CQ overruns.  The local CQs rem_overrun field would need to be checked to determine if the remote CQE landed and if not, the FI_RMA_EVENT message would need to be transferred by other means (like using SMSG as is done for all counter events now).  This implementation would remove the extra round trip required for each transaction when FI_RMA_EVENT is enabled.

**Comment by hppritcha** [comment]
@ztiffany there would need to be a scheme for encoding enough data in the user-data payload of the RX CQE to allow for receiver to discriminate which EP's remote counters to increment.  Also, I think if a target memory region's mr handle hasn't been bound to the EP, and its the target of the RMA operation, then the counter shouldn't be incremented.  But maybe I'm confused on that point.

**Comment by sungeunchoi** [comment]
@hppritcha What do you mean by "mr handle hasn't been bound to the EP"?  Are you talking about a ugni endpoint?  If so, I guess I don't understand what it means in the context of libfabric-level requested remote events.

**Comment by ztiffany** [comment]
@hppritcha good point, I was thinking about the remote event arriving at the SMSG CQ, which would be mapped right to the correct EP (and any bound counters).
